libdnf/rpm/solv/goal_private.cpp

```diff
@@ -396,23 +396,25 @@ libdnf::solv::IdQueue GoalPrivate::list_obsoleted() {
     return list_results(SOLVER_TRANSACTION_OBSOLETED, 0);
 }
 
-void GoalPrivate::write_debugdata(const std::string & dir) {
+void GoalPrivate::write_debugdata(const std::filesystem::path & dir) {
     libdnf_assert_goal_resolved();
 
+    // Ensures the presence of the directory.
+    auto dir_path = std::filesystem::absolute(dir);
+    std::filesystem::create_directories(dir_path);
+
+    // Removes old debug data, if it exists.
+    for (const auto & dir_entry : std::filesystem::directory_iterator(dir_path)) {
+        std::filesystem::remove(dir_entry);
+    }
+
     int flags = TESTCASE_RESULT_TRANSACTION | TESTCASE_RESULT_PROBLEMS;
-    // TODO(jmracek) add support of relative path and create required dirs and parents
-    //     g_autofree char * absdir = abspath(dir);
-    //     if (!absdir) {
-    //         std::string msg = tfm::format(_("failed to make %s absolute"), dir);
-    //         throw Goal::Error(msg, DNF_ERROR_FILE_INVALID);
-    //     }
-    //     makeDirPath(dir);
-    //     g_debug("writing solver debugdata to %s", absdir);
-    auto ret = ::testcase_write(libsolv_solver, dir.c_str(), flags, NULL, NULL);
+    //g_debug("writing solver debugdata to %s", absdir);
+    auto ret = ::testcase_write(libsolv_solver, dir_path.c_str(), flags, NULL, NULL);
 
     if (!ret) {
         // TODO(jmracek) replace error with Goal Error
-        throw RuntimeError(M_("failed writing debugdata"));
+        throw RuntimeError(M_("failed writing debugsolver data into \"{}\""), dir_path.c_str());
```

Comment:
Do you think `pool_errstr()` could contain the description for the error here?

Reply:
I looked in the libsolv source code. And yes `pool_errstr()` should return some error description. I will use it.